src/agent/src/register.cpp

```diff
@@ -0,0 +1,142 @@
+#include "register.hpp"
+#include "configuration_parser/include/configuration_parser.hpp"
+
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/stream.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <iostream>
+#include <jwt-cpp/jwt.h>
+#include <nlohmann/json.hpp>
+
+namespace beast = boost::beast;
+namespace http = beast::http;
+using tcp = boost::asio::ip::tcp;
+using json = nlohmann::json;
+
+namespace registration
+{
+    http::response<http::dynamic_body> sendRequest(const std::string& managerIp,
+                                                   const std::string& port,
+                                                   const http::verb method,
```

Comment:
Is there a reason why the `method` is taken by value instead of by reference?

Reply:
Fixed! There is no reason!